MTA-6811, 6812, 6813, 6814, 6815, 6816 CQA April Developer Lightspeed work (#357)
migtools/mta-documentation: docs/topics/developer-lightspeed/con_developer-lightspeed-logs.adoc

= Debugging {mta-dl-plugin}

[role="_abstract"]
{mta-dl-full} generates logs to debug issues specific to the extension host and the {ProductShortName} analysis and Remote Procedure Call (RPC) server. You can also configure the log level for the {mta-dl-plugin} in the extension settings. The default log level is *debug*.

> {mta-dl-plugin} stores extension logs as `extension.log` with automatic rotation. The maximum size of the log file is 10 MB and {mta-dl-plugin} retains three files. {mta-dl-plugin} stores analyzer RPC logs as `analyzer.log` without rotation.

Reviewer: Since it's not part of the short description, shall we make it a bulleted list? It improves scanability and readability. Just please check if I managed to preserve the technical accuracy when speaking about log file size and number of files.
```suggestion
The {mta-dl-plugin} stores the following logs:

* Extension logs (`extension.log`) with automatic rotation. The plugin retains up to three files with a maximum size of 10 MB each.
* Analyzer RPC logs (`analyzer.log` without automatic rotation.
```

[id="dev-lightspeed-archive-logs_{context}"]

== Archiving the logs

To archive the logs as a zip file, type `{ProductShortName}: Generate Debug Archive` in the Visual Studio Code Command Palette and select the information type to archive as a log file. 

The archive command allows capturing all relevant log files in a zip archive at the specified location in your project. By default, you can access the archived logs in the .vscode directory of your project.

The archival feature helps you to save the following information:

> * Large language model (LLM) provider configuration: Fields from the provider settings that you can include in the archive. {mta-dl-plugin} redacts all fields for security reasons by default. Ensure that you do not expose any secrets.

Reviewer: Reducing wordiness:
```suggestion
* Large language model (LLM) provider configuration: The provider settings fields to include in the archive. To ensure security, the {mta-dl-plugin} redacts all fields by default. Do not expose any secrets.
```

= Debugging {mta-dl-plugin}

[role="_abstract"]
{mta-dl-full} generates logs to debug issues specific to the extension host and the {ProductShortName} analysis and Remote Procedure Call (RPC) server. You can also configure the log level for the {mta-dl-plugin} in the extension settings. The default log level is *debug*.

{mta-dl-plugin} stores the following logs:

* Extension logs (`extension.log`) with automatic rotation. The plugin retails three files with a maximum size of 10 MB each. 

> * Analyzer RPC logs (analyzer.log) without automatic rotation.

Reviewer: ```suggestion
* Analyzer RPC logs (`analyzer.log`) without automatic rotation.
```

You can access the logs in the following ways:

> * *Log file*: Type `Developer: Open Extension Logs Folder` and open the `redhat.mta-vscode-extension` directory that has the extension log and the analyzer log.

Reviewer: ```suggestion
* *Log file*: Type `Developer: Open Extension Logs Folder` and the `redhat.mta-vscode-extension` directory that has the extension log and the analyzer log.
```

Author: Please note that I'm not making this change since the user has to open the `redhat.mta-vscode-extension` directory.
